### ISSUE TITLE
Retry once on send timeout

### ISSUE THREAD

**Issue by JamieDanielson** [honeycombio/libhoney-py#105]
**Is your feature request related to a problem? Please describe.**

Our ALB currently has a timeout of 60s, so idle connections from libhoney that time out around the 60s mark may be dropped.

**Describe the solution you'd like**

Retry once if there is an http timeout, similar to what is done in [libhoney-go](https://github.com/honeycombio/libhoney-go/blob/main/transmission/transmission.go#L386).

**Describe alternatives you've considered**

Considering increasing the ALB timeout to 65s to avoid common 60s intervals, but libhoney transmission should retry once regardless because this issue could technically occur at any timeout interval.

**Comment by emilyashley** [comment]
First thoughts:

If we wanted to use the requests utils it might look something like this when we're getting a request Session

```
        from requests import Session
        from requests.adapters import HTTPAdapter
        from requests.packages.urllib3.util.retry import Retry

        retry_strategy = Retry(total=1, method_whitelist=["POST"]) #allow 1 retry on post
        http_adapter = HTTPAdapter(max_retries=retry_strategy)

        session = Session()
        session.mount("http://", http_adapter) #we could make this more specific to our api endpoint as well
        return session
```

Alternatively, we could add simple retry logic in the POST via _send_batch() in transmission.py similar to the go example linked 🤔 

I'm not sure what to do about the async tornado implementation or what concerns that may bring, worth noting it might be something to break into another ticket if a solution doesn't work for both methods of transmission.

**Comment by kentquirk** [comment]
I wouldn't have even thought to use a library for this -- but this is exactly the kind of thing to use a library for! Someone else has done the hard work (we hope) of debugging this, and I love the idea of leveraging that work. 

An ideal solution would find a way to test that it actually retries the way we want it to, by failing different numbers of times in different tests. And it should probably work with async just fine -- or can be made to.

**Comment by emilyashley** [comment]
Merged in the changes for retrying once during the standard transmission process ✨ However, this solution did _not_ implement a retry for the Tornado async transmission process just yet.

Next steps for me: check out this [Tornado Retry Client]( https://github.com/wpjunior/tornado-retry-client) as either an example to follow or, potentially, a functionality to import. 

Proposed a few changes to improve development on the tornado transmission process, in the meantime.

**Comment by emilyashley** [comment]
Alrighty! I made a few passes and no go. Womp. :(

For what it's worth [the beeline docs](https://docs.honeycomb.io/getting-data-in/beeline/python/) say this about expectations w/r/t python async support:
> The Python Beeline does not currently support asyncio-based frameworks, and has limited support for Tornado. If you are using an asynchronous framework and would like to see support added in the Beeline, please [file an issue](https://github.com/honeycombio/beeline-python/issues)

As such, I think this is good to go without implementing the retry for Tornado. I did add tornado as a transmission method field on our user agent header so we can get a better idea of how often the tornado transmission is used and revisit this retry if desired!
